src/reconcile/rec_write.c

```diff
@@ -521,9 +521,10 @@ __rec_write_page_status(WT_SESSION_IMPL *session, WTI_RECONCILE *r)
          * If the page state changed, the page has been written since reconciliation started and
          * remains dirty (that can't happen when evicting, the page is exclusively locked).
          */
-        if (__wt_atomic_cas32(&mod->page_state, WT_PAGE_DIRTY_FIRST, WT_PAGE_CLEAN))
+        if (__wt_atomic_cas32(&mod->page_state, WT_PAGE_DIRTY_FIRST, WT_PAGE_CLEAN)) {
+            WT_RELEASE_WRITE(mod->modified, (bool)false);
```

Comment:
I suspect some races can leave this page as unmodified, causing the data loss.

1. We mark the page as clean. 
2. Context switch
3. Other thread mark the page as DIRTY_FIRST and set the modified flag as true
4. Now this thread resumes and marks the modified flag as false.

If we don't get any follow-on modifications to this page, this page never gets reconciled, and we can lose the data.

How about rechecking the page state to confirm that it is still clean?

Reply:
yeah, this makes sense but does rechecking solve the issue? this race can still happen after recheck right?

Reply:
Setting the `modified` flag back to true if the page state is not clean after the recheck?

Reply:
Yeah, this is the problem of using two variables for representing a single thing.